python/pyspark/ml/feature.py

```diff
@@ -2869,6 +2869,18 @@ def findSynonyms(self, word, num):
             word = _convert_to_vector(word)
         return self._call_java("findSynonyms", word, num)
 
+    @since("2.2.0")
+    def findSynonymsTuple(self, word, num):
+        """
+        Find "num" number of words closest in similarity to "word".
+        word can be a string or vector representation.
+        Returns an array with two fields word and similarity (which
+        gives the cosine similarity).
+        """
+        if not isinstance(word, basestring):
+            word = _convert_to_vector(word)
+        return self._call_java("findSynonymsTuple", word, num)
+
```

Comment:
We need to convert result back to array of tuple, which would be consistent with Scala output.